### ISSUE TITLE
Building old commit

### ISSUE THREAD

**Issue by tmozgach** [singularityhub/singularityhub.github.io#95]
When  I delete collection and add the same collection again. The Singularity Hub is building old(yesterday commit, Dec. 11, 2017, 3 p.m. commit). When I recommit it. The new commit is stuck in Pending status forever. 
https://www.singularity-hub.org/collections/367
How to correctly stop building and start again? It would be nice to have a button `stop`.

**Comment by vsoch** [comment]
hey @tmozgach ! When you connect a collection for the first time (regardless of whether it existed before) the repository is parsed for all Singularity files, but importantly, only the commits for which you made a change to a file (so if you changed something like a README.md at a later commit but no Singularity files, this commit wouldn't be used for the build). You are only allowed one builder for your collection, so when you have two that need to build (as you do now) the first one has to finish first. So:

1. if you don't want the first to build (RUNNING) you can cancel it. The second that is pending will start.
2. Any commits you want deemed to be "most recent" for a particular Singularity file need to have the Singularity file itself changed.
3. If the commit to build model doesn't fit for you, just change it to be manual in the collection settings,and then you can choose exactly the commit and Singularity files to build.

**Comment by tmozgach** [comment]
@vsoch Thank you for explanation. I did commit exactly for Singularity.1 file today (more the 5 times), but the system starts to build the yesterday commit when I start new collection. I recommitted again and now it took a new commit but stuck in Pending status even if I don't have another buildings. 
Just want to clarify, Cancel function is Trash bin(delete container build) on the left?

**Comment by vsoch** [comment]
yes! I would cancel, then go to build settings and change to manual build, and then build exactly as you want.

**Comment by vsoch** [comment]
Closing issue, collection no longer exists. @tmozgach if you have feedback for the future, we are collecting it here: https://forms.gle/sbb41gWtfLBvmm4k8
